src/Redux/Reducers/authSlice.js

```diff
@@ -0,0 +1,111 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { AlertToast, SuccessToast } from "Components";
+import axios from "axios";
+
+const encodedToken = localStorage.getItem("StormMediaToken");
+const initialState = {
+  currentUser: {},
+};
+
+export const login = createAsyncThunk("auth/login", async (userDetails) => {
+  try {
+    const response = await axios.post(`/api/auth/login`, {
+      username: userDetails.username,
+      password: userDetails.password,
+    });
+    return response.data;
+  } catch (error) {
+    AlertToast(`${error.response.data.errors}`);
+  }
+});
+
+export const signup = createAsyncThunk("auth/signup", async (userDetails) => {
+  try {
+    const response = await axios.post(`/api/auth/signup`, {
+      firstName: userDetails.name,
+      username: userDetails.username,
```

Comment:
Suggestion: same comments as for login, and try to match name of keys similar to backend ( passwordOne as just password ) so mapping can be avoided

Reply:
Thanks for the suggestion will look into it